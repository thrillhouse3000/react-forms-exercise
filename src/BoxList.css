.BoxList {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.BoxList-container {
    margin-left: auto;
    margin-right: auto;
}