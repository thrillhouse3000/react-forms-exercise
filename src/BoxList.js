import {useState} from "react"
import Box from "./Box"
import NewBoxForm from "./NewBoxForm"
import {v4 as uuid} from "uuid"
import './BoxList.css'

const BoxList = () => {
    const INITIAL_STATE = []
    const [boxes, setBoxes] = useState(INITIAL_STATE)

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, {...newBox, id: uuid()}])
    }

    const removeBox = (id) => {
        setBoxes(boxes.filter(b => b.id !== id))
    }

    return (
        <div className='BoxList'>
            <h2>Color Box Maker</h2>
            <NewBoxForm addBox={addBox} />
            <div className='BoxList-container'>
                {boxes.map(b => 
                    <Box key={b.id} id={b.id} color={b.color} width={b.width} height={b.height} removeBox={removeBox}
                />)}
            </div>
        </div>
    )
}

export default BoxList;