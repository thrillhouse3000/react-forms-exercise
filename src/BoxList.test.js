import { render, screen, fireEvent } from '@testing-library/react';
import BoxList from './BoxList';

it('renders without crashing', () => {
    render(<BoxList />)
});

it('matches snapshot', () => {
    const {asFragment} = render(<BoxList />)
    expect(asFragment()).toMatchSnapshot()
});

it('adds a box', () => {
    render(<BoxList />)
    const colorInput = screen.getByLabelText('Color')
    const widthInput = screen.getByLabelText('Width')
    const heightInput = screen.getByLabelText('Height')
    const addBtn = screen.queryByText('Add Box')
    let boxBtn = screen.queryByText('X')
    expect(boxBtn).not.toBeInTheDocument()

    fireEvent.change(colorInput, {target: {value: 'red'}})
    fireEvent.change(widthInput, {target: {value: 2}})
    fireEvent.change(heightInput, {target: {value: 2}})
    fireEvent.click(addBtn)

    boxBtn = screen.queryByText('X')
    expect(boxBtn).toBeInTheDocument()
});

it('deletes a box', () => {
    render(<BoxList />)
    const colorInput = screen.getByLabelText('Color')
    const widthInput = screen.getByLabelText('Width')
    const heightInput = screen.getByLabelText('Height')
    const addBtn = screen.queryByText('Add Box')
    let boxBtn = screen.queryByText('X')
    expect(boxBtn).not.toBeInTheDocument()

    fireEvent.change(colorInput, {target: {value: 'red'}})
    fireEvent.change(widthInput, {target: {value: 2}})
    fireEvent.change(heightInput, {target: {value: 2}})
    fireEvent.click(addBtn)

    boxBtn = screen.queryByText('X')
    expect(boxBtn).toBeInTheDocument()

    fireEvent.click(boxBtn)
    boxBtn = screen.queryByText('X')
    expect(boxBtn).not.toBeInTheDocument()
});