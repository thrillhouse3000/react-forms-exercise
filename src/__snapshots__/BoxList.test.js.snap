// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h2>
      Color Box Maker
    </h2>
    <form>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        type="width"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        type="height"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div
      class="BoxList-container"
    />
  </div>
</DocumentFragment>
`;
